generator/integration_tests/tests/golden_kitchen_sink_stub_factory_test.cc

```diff
@@ -66,6 +69,48 @@ TEST_F(GoldenKitchenSinkStubFactoryTest, DefaultStubWithAuth) {
   EXPECT_THAT(context.credentials(), Not(IsNull()));
 }
 
+#ifdef GOOGLE_CLOUD_CPP_HAVE_OPENTELEMETRY
+
+TEST_F(GoldenKitchenSinkStubFactoryTest, DefaultStubWithTracingEnabled) {
+  using ::google::cloud::testing_util::SpanNamed;
+  auto span_catcher = testing_util::InstallSpanCatcher();
+
+  auto options = Options{}
+                     .set<internal::OpenTelemetryTracingOption>(true)
+                     .set<EndpointOption>("localhost:1");
+  auto stub =
+      CreateDefaultGoldenKitchenSinkStub(CompletionQueue{}, std::move(options));
+  grpc::ClientContext context;
+  (void)stub->DoNothing(context, {});
+
+  auto spans = span_catcher->GetSpans();
+  EXPECT_THAT(
+      spans, Contains(SpanNamed(
+                 "google.test.admin.database.v1.GoldenKitchenSink/DoNothing")));
+}
+
+TEST_F(GoldenKitchenSinkStubFactoryTest, DefaultStubWithTracingDisabled) {
+  using ::google::cloud::testing_util::SpanNamed;
+  using ::testing::Not;
+  auto span_catcher = testing_util::InstallSpanCatcher();
+
+  auto options = Options{}
+                     .set<internal::OpenTelemetryTracingOption>(false)
+                     .set<EndpointOption>("localhost:1");
+  auto stub =
+      CreateDefaultGoldenKitchenSinkStub(CompletionQueue{}, std::move(options));
+  grpc::ClientContext context;
+  (void)stub->DoNothing(context, {});
+
+  auto spans = span_catcher->GetSpans();
+  EXPECT_THAT(
+      spans,
+      Not(Contains(SpanNamed(
+          "google.test.admin.database.v1.GoldenKitchenSink/DoNothing"))));
```

Comment:
Maybe for another PR.  The `/` separator between $service and $RPC does not conform to the gRPC / Protobuf naming conventions.

Reply:
FYI, I am following this: https://opentelemetry.io/docs/reference/specification/trace/semantic_conventions/rpc/#span-name